x-pack/plugins/security/public/management/users/edit_user/change_password_flyout.tsx

```diff
@@ -0,0 +1,286 @@
+              defaultMessage:
+                '{isSubmitting, select, true{Changing password…} other{Change password}}',
+              values: { isSubmitting: form.isSubmitting },
+            })
+      }
+      submitButtonColor={isSystemUser ? 'danger' : undefined}
+      isLoading={form.isSubmitting}
+      isDisabled={isLoading || (form.isSubmitted && form.isInvalid)}
+      size="s"
+      ownFocus
+    >
+      {isLoading ? (
+        <EuiLoadingContent />
+      ) : (
+        <EuiForm component="form" {...eventHandlers}>
+          {isSystemUser ? (
+            <>
+              <EuiCallOut
+                title={i18n.translate(
+                  'xpack.security.management.users.changePasswordFlyout.systemUserTitle',
+                  { defaultMessage: 'This is extremely important!' }
+                )}
+                color="danger"
+                iconType="alert"
+              >
+                <p>
+                  <FormattedMessage
+                    id="xpack.security.management.users.changePasswordFlyout.systemUserWarning"
+                    defaultMessage="Changing this password will prevent Kibana from communicating with Elasticsearch."
+                  />
+                </p>
+                <p>
+                  <FormattedMessage
+                    id="xpack.security.management.users.changePasswordFlyout.systemUserDescription"
+                    defaultMessage="Once changed, you must manually update your config file with the new password and restart Kibana."
+                  />
+                </p>
+              </EuiCallOut>
+              <EuiSpacer />
+            </>
+          ) : undefined}
+
+          <EuiFormRow
+            label={i18n.translate(
+              'xpack.security.management.users.changePasswordFlyout.userLabel',
+              {
+                defaultMessage: 'User',
+              }
+            )}
+          >
+            <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="user" />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiSpacer size="xs" />
+                <EuiText>{username}</EuiText>
+                <EuiSpacer size="xs" />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiFormRow>
+
+          {isCurrentUser ? (
+            <EuiFormRow
+              label={i18n.translate(
+                'xpack.security.management.users.changePasswordFlyout.currentPasswordLabel',
+                { defaultMessage: 'Current password' }
+              )}
+              error={form.errors.current_password}
+              isInvalid={form.touched.current_password && !!form.errors.current_password}
+            >
+              <EuiFieldPassword
+                name="current_password"
+                type="dual"
+                value={form.values.current_password}
```

Comment:
Ah good catch! That's very annoying since I want to use controlled components but rely on event bubbling for the event handlers. I'll just add the event handlers to the input instead of the form to suppress this warning. React trying to be "helpful" 🙄